jetbrains-core/src/software/aws/toolkits/jetbrains/services/dynamic/DynamicResourcesUpdateManager.kt

```diff
@@ -128,29 +129,19 @@ internal class DynamicResourceUpdateManager(private val project: Project) {
 
     private fun getProgress() {
         var size = pendingMutations.size
-
+        var isValidToken = true
         while (size > 0) {
             val mutation = pendingMutations.remove()
             if (!mutation.status.isTerminal()) {
                 val client = mutation.connectionSettings.awsClient<CloudControlClient>()
                 val progress = try {
                     client.getResourceRequestStatus { it.requestToken(mutation.token) }
                 } catch (e: Exception) {
-                    e.notifyError(
-                        message(
-                            "dynamic_resources.operation_status_notification_title",
-                            mutation.resourceIdentifier ?: mutation.resourceType,
-                            mutation.operation.name.toLowerCase()
-                        ),
-                        project
-                    )
-                    DynamicresourceTelemetry.mutateResource(
-                        project,
-                        Result.Failed,
-                        mutation.resourceType,
-                        addOperationToTelemetry(mutation.operation),
-                        ChronoUnit.MILLIS.between(mutation.startTime, DynamicResourceTelemetryResources.getCurrentTime()).toDouble()
-                    )
+                    unableToGetProgress(e, mutation)
```

Comment:
We definitely do not want to mark all exceptions as a failure; it could be a temporary network blip - which we will retry next time around. They need to be handled differently.

Reply:
The exception specific to GetResourceRequestStatus is handled in this case, in most other cases, the operation status will change to failed, which will remove them from the queue with the updated implementation